astropy/units/format/base.py

```diff
@@ -203,8 +200,135 @@ def _to_string(cls, unit: UnitBase, *, fraction: bool | str) -> str:
         )
 
     @classmethod
-    def parse(cls, s: str) -> UnitBase:
+    def parse(cls, s: str, debug: bool = False) -> UnitBase:
```

Comment:
Nitpick
```suggestion
    def parse(cls, s: str, *, debug: bool = False) -> UnitBase:
```

Reply:
I don't think we should be changing the signature of a public method at all in this pull request.

Reply:
Yes, agreed, this is better as just a refactoring.

Reply:
SGTM. For a followup.

Reply:
e.g. https://docs.astral.sh/ruff/rules/boolean-type-hint-positional-argument/

Reply:
The `Base.parse()` method mostly exists to define the public API for downstream packages. Adding a new parameter changes that API and we definitely shouldn't do that in what is supposed to be a refactoring of the private API.

Reply:
IDK. All the other `parse` methods have this optional kwarg. Code shouldn't break for downstream users.